src/snapshot.rs

```diff
@@ -15,6 +15,8 @@ use crate::{
     valve::ValveSystem,
 };
 
+static ALLOWANCE: AtomicU8 = AtomicU8::new(0);

```

Comment:
This doesn't actually need to be a global variable. Nor is it necessary to be atomic. We should use a regular local `u8` variable.